src/generators/java/renderers/EnumRenderer.ts

```diff
@@ -0,0 +1,70 @@
+import { JavaRenderer } from "../JavaRenderer";
+
+import { FormatHelpers } from "../../../helpers";
+
+/**
+ * Renderer for Java's `enum` type
+ * 
+ * @extends JavaRenderer
+ */
+export class EnumRenderer extends JavaRenderer {
+  render(): string {
+    const enumName = this.model.$id;
+    const type = Array.isArray(this.model.type) ? this.model.type[0] : this.model.type!;
+
+    switch(type) {
+      case "string": {
+        return `public enum ${enumName} {
+${this.indent(this.renderItems("String"))};
+
+${this.indent(this.renderHelpers(enumName!, "String"))}
+}`;
+      } 
+      case "integer": {
+        return `public enum ${enumName} {
+${this.indent(this.renderItems("Integer"))};
+
+${this.indent(this.renderHelpers(enumName!, "Integer"))}
+}`;
+      }
+      default: return "";
+    }
+  }
+
+  protected renderItems(type: "String" | "Integer" = "String"): string {
+    const enums = this.model.enum || [];
+    const items = enums.map(e => {
+      const name = FormatHelpers.toConstantCase(type === "String" ? e : `number ${e}`);
+      return `${name}("${e}")`;
+    }).join(", ");
+    return `${items}`;
+  }
+
+  protected renderHelpers(enumName: string, type: "String" | "Integer" = "String"): string {
```

Comment:
Why only `"String" | "Integer"` ?

Reply:
As I see in examples in net, Java cannot handle enums like in TS by `{key} = {value}` syntax, and also you can specify only some primitives in enum like String, Integer (I remember you can also use Boolean).

Reply:
You can with [constructors](https://stackoverflow.com/questions/40547499/how-to-make-key-value-like-enum-in-java) 😄

Reply:
I also used constructors :P but then you have more complex type. QuickType for example, create enums only for strings, and then if you have defined strings, boolean, integers in enums, QT creates extra type (and treat it as union) to handle boolean and integers :P